opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -1414,28 +1415,35 @@ void EclipseWriter::writeTimeStep(const SimulatorTimerInterface& timer,
 }
 
 
-EclipseWriter::EclipseWriter(const parameter::ParameterGroup& params,
-                             Opm::EclipseStateConstPtr eclipseState,
-                             const Opm::PhaseUsage &phaseUsage,
-                             int numCells,
-                             const int* compressedToCartesianCellIdx)
+EclipseWriter::EclipseWriter(Opm::EclipseStateConstPtr eclipseState,
+                             const Opm::PhaseUsage &phaseUsage)
     : eclipseState_(eclipseState)
-    , numCells_(numCells)
-    , compressedToCartesianCellIdx_(compressedToCartesianCellIdx)
-    , gridToEclipseIdx_(numCells, int(-1) )
     , phaseUsage_(phaseUsage)
 {
     const auto eclGrid = eclipseState->getInputGrid();
     cartesianSize_[0] = eclGrid->getNX();
     cartesianSize_[1] = eclGrid->getNY();
     cartesianSize_[2] = eclGrid->getNZ();
 
-    if( compressedToCartesianCellIdx ) {
+    numCells_ = eclGrid->getNumActive();
+    gridToEclipseIdx_ = std::vector<int>(numCells_, int(-1) );
+
+    //TODO
+    //This should be calculated in EclipseGrid
+    std::shared_ptr<Opm::GridManager>  ourFineGridManagerPtr = std::make_shared<Opm::GridManager>(eclipseState->getInputGrid());
```

Comment:
This is quite dangerous. You are building a copy of the grid inside the writer, whereas the outside could possibly use a different mechanism to build the grid (for example flow_mpi.cpp uses the CpGrid class). That could lead to inconsistencies and it can also be a bit expensive. I think that the necessary parts must still be passed in the constructor of EclipseWriter, as before.